okhttp-tests/src/test/java/okhttp3/CallTest.java

```diff
@@ -1137,7 +1137,8 @@ private void postBodyRetransmittedAfterAuthorizationFail(String body) throws Exc
 
     client = client.newBuilder()
         .hostnameVerifier(new RecordingHostnameVerifier())
-        .connectionSpecs(Arrays.asList(ConnectionSpec.MODERN_TLS, ConnectionSpec.COMPATIBLE_TLS))
+        // Attempt RESTRICTED_TLS then fall back to MODERN_TLS.
+        .connectionSpecs(Arrays.asList(ConnectionSpec.RESTRICTED_TLS, ConnectionSpec.MODERN_TLS))
```

Comment:
Is this a good point for a varargs method on connectionSpecs?

Reply:
I don’t mind this API being a little awkward to use. Most users shouldn’t use it!